src/global/pidfile.cc

```diff
@@ -96,3 +96,27 @@ int pidfile_remove(void)
   pid_file[0] = '\0';
   return 0;
 }
+
+int test_pid_file_in_use(const md_config_t *conf) 
+{
+  int fd;
+  if (conf->pid_file.empty()) {
+    return pidfile_remove();
```

Comment:
the `pid_file` variable is initialized by `pidfile_write()`, which is called **after** `test_pid_file_in_use()` gets called. so `pidfile_remove()` always returns 0 here?